### DISCUSSION TITLE
Unable to get styles to apply in react-native web

### DISCUSSION THREAD

**Discussion by nickfortyau** [nativewind/nativewind#268] [Q&A] [answered]
I have a react-native application. I'm adding tailwind css via the `className` property to a *.tsx component file. The tailwind styles I added are **showing up correctly in mobile**. However, **in web, none of the styles are rendering**. Is there a caveat to using webpack? Any additional configuration needed to get the web version of the app to recognize tailwind styles?

I've tried the `--reset-cache` argument when launching metro. Tried targeting all possible paths in the tailwind.config.js, even tried a global `./**/*.{html,js}`.

Here's a quick example of my usage in a component at path `./src/components/AccountBalance.tsx`:
```
import { Text } from 'react-native-paper'

export default function AccountBalance() {
  return (
    <Text className="text-green-500">I'm some text</Text>
  )
}
```

tailwind.config.js
```
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    "./web/index.html",
    "./index.web.js",
    "./index.js",
    "./src/App.{js,jsx,ts,tsx}",
    "./src/components/**/*.{js,jsx,ts,tsx}",
    "./src/components/**/*.web.{js,jsx,ts,tsx}"],
  theme: {
    extend: {},
  },
  plugins: [],
}
```

babel.config.js
```
module.exports = {
  presets: ['module:metro-react-native-babel-preset'],
  plugins: ['nativewind/babel', 'react-native-reanimated/plugin']
}
```

**Comment by marklawlor** [comment, marked as answer]
NativeWind does not compile the styles on web, it simply is a wrapper around React Native Web to make classNames work. You need to setup Tailwind CSS like your would any other web project. If you are using Expo there is an example project in the repo

**Reply by nickfortyau** [reply]
Aha! That makes total sense. Thanks for the prompt response, Mark!

**Reply by SkroooB** [reply]
Can you send the link of the expo example project repo

**Reply by marklawlor** [reply]
https://github.com/marklawlor/nativewind/tree/main/examples/expo

**Reply by benjaminb10** [reply]
Hi, is there a documentation on file to update or add to add tailwind for web on a RN project using nativewind?

**Reply by corysimmons** [reply]
https://www.nativewind.dev/quick-starts/expo#expo-web You have to follow it and compare it against the example project here https://github.com/marklawlor/nativewind/tree/main/examples/expo 

I do kinda wish this was somehow bundled into NativeWind core though.

**Reply by corysimmons** [reply]
Actually I can get the Expo example to work, but it doesn't automatically generate the classes for web that I want to use. That is, if I change `bg-red-500` to `bg-yellow-500`, it will hot reload on iOS Simulator, but on web it will just remove the bg color.

If I save `postcss.config.js` it hot reloads.

I'm guessing this has something to do with the webpack config and `postcss-loader` not running on `.tsx` changes (like changing `<Text className="bg-red-500">` to `<Text className="bg-yellow-500">`) but I can't figure it out right now.

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
